dht.go

```diff
@@ -404,24 +415,47 @@ func (dht *IpfsDHT) putLocal(key string, rec *recpb.Record) error {
 }
 
 // peerFound signals the routingTable that we've found a peer that
-// supports the DHT protocol.
-func (dht *IpfsDHT) peerFound(ctx context.Context, p peer.ID) {
-	logger.Event(ctx, "peerFound", p)
-	dht.routingTable.HandlePeerAlive(p)
+// might support the DHT protocol.
+func (dht *IpfsDHT) peerFound(ctx context.Context, p peer.ID, queryPeer bool) {
+	b, err := dht.validRTPeer(p)
+	if err != nil {
+		logger.Errorf("failed to validate if peer is a DHT peer, err=%s", err)
+	} else if b {
+		logger.Event(ctx, "peerFound", p)
+		dht.routingTable.TryAddPeer(p, queryPeer)
+		if queryPeer {
+			dht.routingTable.UpdateLastSuccessfulOutboundQuery(p, time.Now())
+		}
```

Comment:
This isn't necessary anymore right?

Reply:
The thing is:

1. Before there is a query, there is a connection.
2.If a peer ALREADY exists in the Routing Table, we don't do anything for idempotence.

So, if we don't do this, we will end up a "zero time" in the bucket for peers that connect and then query.